packages/ember-application/lib/system/engine-instance.js

```diff
@@ -0,0 +1,79 @@
+/**
+@module ember
+@submodule ember-application
+*/
+
+import EmberObject from 'ember-runtime/system/object';
+import Registry from 'container/registry';
+import ContainerProxy from 'ember-runtime/mixins/container_proxy';
+import RegistryProxy from 'ember-runtime/mixins/registry_proxy';
+import { get } from 'ember-metal/property_get';
+import { set } from 'ember-metal/property_set';
+import run from 'ember-metal/run_loop';
+
+/**
+  The `EngineInstance` encapsulates all of the stateful aspects of a
+  running `Engine`.
+
+  @public
+  @class Ember.EngineInstance
+  @extends Ember.Object
+  @uses RegistryProxyMixin
+  @uses ContainerProxyMixin
+*/
+
+let EngineInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
+  /**
+    The base `Engine` for which this is an instance.
+
+    @property {Ember.Engine} engine
+    @private
+  */
+  base: null,
+
+  init() {
+    this._super(...arguments);
+
+    let base = get(this, 'base');
+
+    if (!base) {
+      base = get(this, 'application');
+      set(this, 'base', base);
```

Comment:
I can do that if I modify some `ApplicationInstance` tests, which I was trying to avoid. However, it's probably safe to do, since every real world `ApplicationInstance` will be instantiated from its associated `Application`.